Library/Homebrew/test/testing_env.rb

```diff
@@ -20,6 +20,22 @@
   abort "Run `bundle install` or install the mocha and minitest gems before running the tests"
 end
 
+if ENV["HOMEBREW_TESTS_COVERAGE"]
+  # Ensure all Ruby files are `require`d for an accurate coverage measurement
+  library_base = File.expand_path("../..", __FILE__)
+  Dir["#{library_base}/**/*.rb"].each do |f|
+    f_rel = f[(library_base.length + 1)..-1] # path relative to `library_base`
+    # scripts
+    next if %w[build.rb postinstall.rb test.rb].include?(f_rel)
+    # mocked files
+    next if %w[config.rb global.rb].include?(f_rel)
+    # test files
+    next if f_rel.start_with?("test/")
```

Comment:
Maybe. But I guess the `add_filter "Homebrew/compat/"` and the `--no-compat` flag of `brew tests` is there for a reason?